site/content/integrate/apps/quick-start-serverless/_index.md

```diff
@@ -0,0 +1,443 @@
+								"path": "/send"
+							}
+						}
+					]
+				}
+			]
+		}
+	]
+}
+```
+
+#### `send_form.json`
+
+```json
+{
+	"type": "form",
+	"form": {
+		"title": "Hello, serverless!",
+		"icon": "/static/icon.png",
+		"fields": [
+			{
+				"type": "text",
+				"name": "message",
+				"label": "message"
+			}
+		],
+		"call": {
+			"path": "/send"
+		}
+	}
+}
+```
+
+### `function.go`
+
+```go
+package function
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/apps/appclient"
+	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
+)
+
+//go:embed bindings.json
+var bindingsData []byte
+
+//go:embed send_form.json
+var formData []byte
+
+// Handler for OpenFaaS and faasd.
+func Handle(w http.ResponseWriter, r *http.Request) {
+	InitApp(apps.DeployOpenFAAS)
+	http.DefaultServeMux.ServeHTTP(w, r)
+}
+```
+
+Note: names `package function` and `func Handle` appear to be required
+(hardcoded) for OpenFaaS.
+
+`golang-middleware` template makes use of adding boilerplate handlers to
+`http.DefaultServeMux`, so we just need to add our handlers and serve.
+
+```go
+var deployType apps.DeployType
+
+func InitApp(dt apps.DeployType) {
+	// Returns the Channel Header and Command bindings for the App.
+	http.HandleFunc("/bindings", httputils.HandleJSONData(bindingsData))
+
+	// The form for sending a Hello message.
+	http.HandleFunc("/send/form", httputils.HandleJSONData(formData))
+
+	// The main handler for sending a Hello message.
+	http.HandleFunc("/send/submit", send)
+
+	deployType = dt
+}
+
+func send(w http.ResponseWriter, req *http.Request) {
+	creq := apps.CallRequest{}
+	json.NewDecoder(req.Body).Decode(&creq)
+
+	message := fmt.Sprintf("Hello from a serververless app running as %s!", deployType)
+	v, ok := creq.Values["message"]
+	if ok && v != nil {
+		message += fmt.Sprintf(" ...and %s!", v)
+	}
+
+	asBot := appclient.AsBot(creq.Context)
+	asBot.DM(creq.Context.ActingUserID, message)
+
+	httputils.WriteJSON(w,
+		apps.NewOKResponse(nil, "Created a post in your DM channel."))
+}
+```
+
+#### `go.mod` and `go.sum`
+
+Since package `function` is built independently by OpenFaaS, add its own `go.mod` for its ddependencies:
+
+```go
+module {{path-to-your-module}}/function
+
+go 1.16
+
+require github.com/mattermost/mattermost-plugin-apps v0.8.0
+```
+
+### 4. `./aws`
+#### `main.go`
+```go
+package main
+
+import (
+    "net/http"
+
+    "github.com/aws/aws-lambda-go/lambda"
+    "github.com/awslabs/aws-lambda-go-api-proxy/httpadapter"
+    "github.com/mattermost/mattermost-plugin-apps/apps"
+
+    "{{path-to-your-module}}/function"
+)
+
+func main() {
+    function.InitApp(apps.DeployAWSLambda)
+
+    // run with the Lambda request re-marshaled back into an http.Request so that it works with mux.
+    lambda.Start(httpadapter.New(http.DefaultServeMux).Proxy)
+}
+```
+
+#### `go.mod`
+```go
+module {{path-to-your-module}}/aws
```

Comment:
Same question here

Reply:
For `main`'s a common top-level `go.mod` may be ok.